src/Utils/utils.php

```diff
@@ -137,26 +136,69 @@ function get_time_format(): string {
 }
 
 /**
- * Gets number in formatted form i.e. express bigger numbers in form of thousands (K), millions (M), billions (B).
+ * Gets number in formatted form i.e. express bigger numbers in form of
+ * thousands (k), millions (M), billions (B).
+ *
+ * Note: This function is not made to process float numbers, and it is a PHP
+ * port of our formatToImpreciseNumber() TypeScript function.
  *
  * Example:
  *   - Represent 10000 as 10K.
  *
  * @since 3.7.0
  *
- * @param int|float $number Number that we have to format.
+ * @param string $value           The number to process. It can be formatted.
+ * @param int    $fraction_digits The number of desired fraction digits.
+ * @param string $glue            A string to put between the number and unit.
  *
- * @return string
+ * @return string The number formatted as an imprecise number.
  */
-function get_formatted_number( $number ): string {
-	$number_formatter = new NumberFormatter( 'en', NumberFormatter::PADDING_POSITION );
-	$formatted_number = $number_formatter->format( $number );
+function get_formatted_number( string $value, int $fraction_digits = 1, string $glue = '' ): string {
```

Comment:
The implementation of this function seems complex can we make it something like following which is an easier implementation?

```php
function formatNumber($number) {
    $suffixes = ["", "K", "M", "B", "T", "Q"];
    $index = 0;

    while ($number >= 1000 && $index < count($suffixes) - 1) {
        $number = $number / 1000;
        $index++;
    }

    return round($number, 1) . $suffixes[$index];
}

// Example usage:
echo formatNumber(10000); // Output: 10K
echo formatNumber(1500000); // Output: 1.5M
echo formatNumber(2500000000); // Output: 2.5B
echo formatNumber(3500000000000); // Output: 3.5T
echo formatNumber(4500000000000000); // Output: 4.5Q
```

Reply:
I agree this is complicated. This is the way Parse.ly does imprecise numbers on the dashboard, so the same approach got used for consistency and for not increasing support questions like "why do I see number `x` on the Post List Stats, but number `y` in the Parse.ly dash?".

I'm more than happy to use the simplified version if it proves to have the same output as the Parse.ly dash, which would need writing related tests on our end. The focus of this PR is to just drop the dependency so I'd leave things as they are, but I'm happy to open an issue for going towards a simplification and a verification that our numbers match the Parse.ly dash.

Reply:
If Parse.ly code is in PHP then its fine but if its not then I think we should take on the refactor now as this is not a huge effort.

Reply:
The original function is in JS and was ported to typeScript in our code. Then I ported the TS code to PHP. Replacing the function isn't a lot of work, but testing that our numbers match those of the Parse.ly dash and writing tests for that is more work.

Reply:
Okay approving the PR then but feel free to take another opinion.